src/mapping.rs

```diff
@@ -0,0 +1,313 @@
+//! A Parser for Proguard Mapping Files.
+//!
+//! The mapping file format is described
+//! [here](https://www.guardsquare.com/en/products/proguard/manual/retrace).
+
+#[cfg(feature = "uuid")]
+use uuid::Uuid;
+
+/// A Proguard Mapping file.
+pub struct ProguardMapping<'s> {
+    source: &'s [u8],
+}
+
+impl<'s> ProguardMapping<'s> {
+    /// Create a new Proguard Mapping.
+    pub fn new(source: &'s [u8]) -> Self {
+        Self { source }
+    }
+
+    /// Calculates the UUID of the mapping file.
+    #[cfg(feature = "uuid")]
+    pub fn uuid(&self) -> Uuid {
+        let namespace = Uuid::new_v5(&Uuid::NAMESPACE_DNS, b"guardsquare.com");
+        // this internally only operates on bytes, so this is safe to do
+        Uuid::new_v5(&namespace, self.source)
+    }
+
+    /// Create an Iterator over [`MappingRecord`]s.
+    ///
+    /// [`MappingRecord`]: enum.MappingRecord.html
+    pub fn iter(&self) -> MappingRecordIter<'s> {
+        MappingRecordIter { slice: self.source }
+    }
+}
+
+/// Split the input `slice` on line terminators.
+///
+/// This is basically [`str::lines`], except it works on a byte slice.
+/// Also NOTE that it does not treat `\r\n` as a single line ending.
+fn split_line(slice: &[u8]) -> (&[u8], &[u8]) {
+    for (i, c) in slice.iter().enumerate() {
+        if *c == b'\n' || *c == b'\r' {
+            return (&slice[0..i], &slice[i + 1..]);
+        }
+    }
+    (slice, &[])
+}
+
+/// An Iterator yielding [`MappingRecord`]s.
+///
+/// [`MappingRecord`]: enum.MappingRecord.html
+pub struct MappingRecordIter<'s> {
```

Comment:
Please implement `Debug` and `Clone`. You have two options:
- Just print the name of this type
- Clone internally, iterate and print the records.

Also, please derive `Default`, which can be extremely useful when using this in higher-level iterators or structs.